engine/src/flutter/impeller/geometry/path_builder.cc

```diff
@@ -219,6 +270,120 @@ PathBuilder& PathBuilder::AddRoundRect(RoundRect round_rect) {
   return *this;
 }
 
+PathBuilder& PathBuilder::AddRoundSuperellipse(RoundSuperellipse rse) {
+  if (rse.IsRect()) {
+    return AddRect(rse.GetBounds());
+  }
+
+  auto SuperellipseArcPoints = [](Point* output,
```

Comment:
You've got a lot of closures closing over other closures, that makes this code quite hard to follow. id commend a class with some methods, i.e. call it something like RoundSuperEllipseBuilder.